### ISSUE TITLE
Memory leak in frame subscription

### ISSUE THREAD

**Issue by aeh** [atom/electron#3401]
Hi Guys,

I've been experimenting with recording an electron sessions by using frame subscriptions and then piping those frames through to ffmpeg.  That seems to work quite well however electron chews through the memory at a rather scary rate.  With a simple css animation at a screen size of 720p it grows by ~8G a minute.

I'm on OSX 10.10.5

```
electron --version
v0.34.3
```

Here is some somewhat contrived code that shows what I'm doing...
```javascript
// index.js
var app = require('app'),
    BrowserWindow = require('browser-window');

app.on('ready', function() {
  var mainWindow = new BrowserWindow({ width: 1280, height: 720 });
  mainWindow.loadUrl('file://' + __dirname + '/index.html');
  mainWindow.webContents.on('did-finish-load', function() {
    mainWindow.webContents.beginFrameSubscription(function() {
      // ...
    });
  });
});
```

```html
<!-- index.html -->
<html>
  <head>
    <style type="text/css">
      @keyframes animation {
        0% { width: 0 }
        100% { width: 100%; }
      }
      .animated {
        position: absolute;
        height: 5px;
        background: red;
        animation: animation 1s infinite linear;
      }
    </style>
  </head>
  <body>
    <div class="animated"></div>
  </body>
</html>
```

**Comment by zcbenz** [comment]
It has been fixed by #3441.

**Comment by brian-mann** [comment]
@aeh I know this is really old but can you show an example of piping those frames through to `ffmpeg`? That is exactly what I'm trying to do.
